From 745f0df47b534e0b1a18ba4f88660c61edc77145 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Mon, 26 Jan 2015 20:19:01 -0500
Subject: [fs] NFSv4.1: Fix an Oops in nfs41_walk_client_list

Message-id: <1422303541-3230-1-git-send-email-steved@redhat.com>
Patchwork-id: 103572
O-Subject: [RHEL7.1 PATCH 1/1][bz1185784] NFSv4.1: Fix an Oops in nfs41_walk_client_list
Bugzilla: 1185784
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Sachin Prabhu <sprabhu@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

If we start state recovery on a client that failed to initialise correctly,
then we are very likely to Oops.

Reported-by: "Mkrtchyan, Tigran" <tigran.mkrtchyan@desy.de>
Link: http://lkml.kernel.org/r/130621862.279655.1421851650684.JavaMail.zimbra@desy.de
Cc: stable@vger.kernel.org
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id: 3175e1dcec40fab1a444c010087f2068b6b04732
Bugzilla: 1185784

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/nfs4client.c b/fs/nfs/nfs4client.c
index ffdb28d..4a5c6e0 100644
--- a/fs/nfs/nfs4client.c
+++ b/fs/nfs/nfs4client.c
@@ -646,7 +646,7 @@ int nfs41_walk_client_list(struct nfs_client *new,
    prev = pos;
 
    status = nfs_wait_client_init_complete(pos);
-   if (status == 0) {
+   if (pos->cl_cons_state == NFS_CS_SESSION_INITING) {
     nfs4_schedule_lease_recovery(pos);
     status = nfs4_wait_clnt_recover(pos);
    }
-- 
1.7.1